### PR TITLE
parallel-benchmark: Allow comparing against older versions

### PR DESCRIPTION
**Pull request by def-**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: test/parallel-benchmark/mzcompose.py [hunk 1]
```diff
@@ -1297,14 +1104,189 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
...
+                            test_class_name_override=scenario,
+                        )
+                    )
+
+    return failures
+
+
+def resolve_tag(tag: str) -> str:
+    if tag == "common-ancestor":
+        # TODO: We probably will need overrides too
```
Comment (nrainer-materialize) [inline test/parallel-benchmark/mzcompose.py:1169]: Yeah, that will likely be needed.

File: test/parallel-benchmark/mzcompose.py [hunk 2]
```diff
@@ -1297,14 +1104,189 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
...
+) -> list[TestFailureDetails]:
+    failures: list[TestFailureDetails] = []
+
+    assert len(this_stats) == len(other_stats)
+
+    for scenario in this_stats.keys():
+        for query in this_stats[scenario].keys():
+            for stat in dir(this_stats[scenario][query]):
+                if stat not in REGRESSION_THRESHOLDS:
+                    continue
```
Comment (nrainer-materialize) [inline test/parallel-benchmark/mzcompose.py:1138]: How can this happen?

File: test/parallel-benchmark/mzcompose.py [hunk 3]
```diff
@@ -1297,14 +1104,189 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
...
+                    # TODO: Table instead?
+                    failure = f"Scenario {scenario} regressed: {query}: {stat}: {this_value:.2f} {'<' if less_than else '>'} {other_value:.2f} ({percent:.2f}% regression)"
+                    print(failure)
+                    failures.append(
+                        TestFailureDetails(
+                            message=failure,
+                            details=str(this_value),
+                            test_class_name_override=scenario,
+                        )
+                    )
```
Comment (nrainer-materialize) [inline test/parallel-benchmark/mzcompose.py:1162]: Optional: Candidate(s) for extracting methods

File: test/parallel-benchmark/mzcompose.py [hunk 4]
```diff
@@ -1297,14 +1104,189 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
...
+                    if less_than
+                    else this_value > other_value * REGRESSION_THRESHOLDS[stat]
+                ):
+                    percent = (
+                        ((other_value - this_value) / other_value)
+                        if less_than
+                        else ((this_value - other_value) / (other_value))
+                    ) * 100
+                    # TODO: Table instead?
+                    failure = f"Scenario {scenario} regressed: {query}: {stat}: {this_value:.2f} {'<' if less_than else '>'} {other_value:.2f} ({percent:.2f}% regression)"
```
Comment (nrainer-materialize) [inline test/parallel-benchmark/mzcompose.py:1154]: Do we also want to print significant improvements?

**Review by nrainer-materialize — APPROVED** [review]

File: test/parallel-benchmark/mzcompose.py (hunk 2 above)
Comment (def-) [reply test/parallel-benchmark/mzcompose.py:1138]: Oops, that should have been a `is None`

File: test/parallel-benchmark/mzcompose.py (hunk 4 above)
Comment (def-) [reply test/parallel-benchmark/mzcompose.py:1154]: Yes, that's why I wanted to make it a table instead.

**Pull request merged**